File: 2107-find-unique-binary-string/2107-find-unique-binary-string.erl
-spec find_different_binary_string(Nums :: [unicode:unicode_binary()]) -> unicode:unicode_binary().
find_different_binary_string(Nums) ->
    Set = sets:from_list(lists:map(fun binary_to_list/1, Nums)),
    N = length(Nums),
    list_to_binary(bcktrk([$0, $1], N, Set, 0, [])).

bcktrk(_, N, Set, N, Tmp) ->
    case sets:is_element(Tmp, Set) of
        true -> [];
        false -> Tmp
    end;
bcktrk(List, N, Set, Len, Tmp) ->
    lists:foldl(
        fun(X, Res) ->
            _Res = bcktrk(N, Set, Len + 1, Tmp ++ [X]),
            case _Res of
                [] -> Res;
                _ -> _Res
            end
        end, [], List
    ).

File: 2107-find-unique-binary-string/2107-find-unique-binary-string.cpp
class Solution {
public:
    void bcktrk(const int n, unordered_set<string>& s, string& tmp, string* res){
        if(tmp.length() == n){
            if(!s.contains(tmp)){
                (*res) = tmp;
            }
            return;
        }

        for(char c = '0'; c <= '1'; ++c){
            tmp += c;
            bcktrk(n, s, tmp, res);
            tmp = tmp.substr(0, tmp.length() - 1);
            if((*res) != "") return;
        }
    }

    string findDifferentBinaryString(vector<string>& nums) {
        const int n = nums.size();
        unordered_set<string> s{};

        for(int i = 0; i < n; ++i)
            s.insert(nums[i]);

        string ans, tmp;
        bcktrk(n, s, tmp, &ans);
        return ans;        
    }
};